src/astUtils/AstEditor.ts

```diff
@@ -0,0 +1,107 @@
+export class AstEditor {
+    private changes: Change[] = [];
+
+    /**
+     * Indicates whether the editor have changes that were applied
+     */
+    public get hasChanges() {
+        return this.changes.length > 0;
+    }
+
+    /**
+     * Change the value of an object's property
+     */
+    public setProperty<T, K extends keyof T>(obj: T, key: K, newValue: T[K]) {
+        const change = new EditPropertyChange(obj, key, newValue);
+        this.changes.push(change);
+        change.apply();
+    }
+
+    /**
+     * Insert an element into an array at the specified index
+     */
+    public addToArray<T extends any[]>(array: T, index: number, newValue: T[0]) {
+        const change = new AddToArrayChange(array, index, newValue);
+        this.changes.push(change);
+        change.apply();
+    }
+
+    /**
+     * Remove an element from an array at the specified index
+     */
+    public removeFromArray<T extends any[]>(array: T, index: number) {
+        const change = new RemoveFromArrayChange(array, index);
+        this.changes.push(change);
+        change.apply();
+    }
+
+    /**
+     * Unto all changes.
+     */
+    public undoAll() {
```

Comment:
`undoAll` shouldn't be in the API exposed to plugins

Reply:
Fair point. I updated the interface for the transpile events: 

https://github.com/rokucommunity/brighterscript/blob/c617cc98302c968c32ac6369c67c28a882d8547e/src/interfaces.ts#L252